packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -399,7 +403,29 @@ export function TransferPanel() {
 
             await latestToken.current.approve({
               erc20L1Address: selectedToken.address,
-              l1Signer: latestNetworksAndSigners.current.l1.signer
+              l1Signer: latestNetworksAndSigners.current.l1.signer,
+              txLifecycle: {
+                onTxSubmit: ({ tx }) => {
+                  transactions.addTransaction({
```

Comment:
imo the file can be cleaned up by abstracting some code into custom hooks
again, for the purpose of this PR, I would think it's out of scope to make those changes
otherwise this PR can grow exponentially and will be unreviewable